langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java

```diff
@@ -111,4 +111,82 @@ void should_allow_custom_generativeModel_and_generationConfig() throws IOExcepti
         // then
         assertThat(response.content().text()).contains("Berlin");
     }
-}
\ No newline at end of file
+
+
+    @Test
+    void should_maintain_a_conversation() {
+
+        // given
+        ChatLanguageModel model = VertexAiGeminiChatModel.builder()
+            .project("langchain4j")
+            .location("us-central1")
+            .modelName("gemini-pro")
+            .maxOutputTokens(1024)
+            .build();
+
+        ConversationalChain chain = ConversationalChain.builder()
+            .chatLanguageModel(model)
+            .build();
+
+        // when/then
+        String resp = chain.execute("What color is the sky usually?");
+        assertThat(resp).containsIgnoringCase("blue");
+
+        resp = chain.execute("Why is it of that color?");
+        assertThat(resp).containsIgnoringCase("scattering");
+
+        resp = chain.execute("What's the name of the scattering effect?");
+        assertThat(resp).containsIgnoringCase("Rayleigh");
+    }
+
+    static class Person {
+        String name;
+        int age;
+    }
+
+    interface PersonExtractor {
+        @dev.langchain4j.service.UserMessage(value = {
+            "Extract the name and age of the person described below.",
+            "Return a JSON document with a \"name\" and an \"age\" property, following this structure: ",
```

Comment:
The idea of `AiServices` is that user does not need to specify this detailed information on how and what to return (formatting).
User just defines method return type and `AiServices` takes care of the rest (under the hood, it automatically creates and provides json schema to the model).

So I would suggest to simplify this to:
```
interface PersonExtractor {
        @dev.langchain4j.service.UserMessage("Extract information about a person from {{it}}")
        Person extractPerson(String text);
    }
```

You will also need to add this to `pom.xml` to make it compile:
```
        <dependency>
            <groupId>dev.langchain4j</groupId>
            <artifactId>langchain4j</artifactId>
            <scope>test</scope>
        </dependency>
```